src/System/Button.cpp

```diff
@@ -0,0 +1,35 @@
+#include "Button.h"
+
+#define DEBOUNCE_TIME 50
+
+Button::Button(uint16_t pin, bool activeHigh, void (*callbackPosEdge)() = NULL, void (*callbackNegEdge)() = NULL){
+    _pin = pin;
+    _activeHigh = activeHigh;
+    _callbackPosEdge = callbackPosEdge;
+    _callbackNegEdge = callbackNegEdge;
+
+    // Enable input pin
+    if(_activeHigh){
+        pinMode(pin, INPUT_PULLDOWN);
+    }else{
+        pinMode(pin, INPUT_PULLUP);
+    }
+}
+
+bool Button::handle(){
+    bool newState = digitalRead(_pin);
+    if(_activeHigh){
+        newState = !newState;
```

Comment:
I'm a bit confused by the logic here.  Why negate the newState bool when activeHigh is true?

Reply:
Tried to make another comment bit github glitched out or something.

I think the semantics of the variable names might be skewing things a bit.  In button.h you comment that activeHigh = true means the button is 'on' for logical high.  I would assume that means when you push the button, digitalRead returns HIGH.  If i've somehow misinterpreted that then you can ignore this next part:

So in the case of the button you defined in main, if I push the button, newState will be inverted to false, and the fn will attempt to call callbackNegEdge, which is null.  So it's only when I release the button after holding it for 50ms that buttonPushed will actually be called.  Is that the idea?

Reply:
Ok so you are 100% right, however for some reason the functionality is backwards from what is expected. I will have to probe with a multimeter when I have a chance and see WTF is going on.

Reply:
LOL I figured it out, the button I bought is "normally-closed" which means pushing it actually disconnects it. I'm stupid!

Reply:
hah!  I also feel like it might make more sense to name the member function pointers 'callbackButtonPushed' and 'callbackButtonReleased' to reduce confusion between digitalRead, newState and button behavior, at least for the non-electrically savvy like me

Reply:
Done!